tests/parsenum/main.c

```diff
@@ -16,16 +16,36 @@ main(int argc, char * argv[])
 
 	WARNP_INIT;
 
-#define TEST(x, y, z, w) do {						\
-	fprintf(stderr, "PARSENUM(\"%s\", %s, %s)\n", y, #z, #w);	\
-	if (PARSENUM(x, y, z, w))					\
-		warnp("PARSENUM");					\
-	fprintf(stderr, "%f %zu %d\n", d, s, i);			\
+#define TEST4_PROLOGUE(str, var, min, max, target)		\
```

Comment:
Either copy the lines, or pick a different name.  A prologue is something which isn't really part of the routine itself, e.g. fiddling with the stack pointer: https://en.wikipedia.org/wiki/Function_prologue .  What you have here is the the first half of the test (in the sense that a test is 1. running a command 2. checking the results).

BTW you probably want to adjust how you organize the macros anyway, since the check-the-results bits for PARSENUM4 and PARSENUM2 should be the same...